adr/0251-use-catalyst-for-client-side-behaviours.md

```diff
@@ -0,0 +1,28 @@
+# 251. Use Catalyst for client-side behaviours
+
+Date: 2022-04-12
+
+## Status
+
+Accepted
+
+## Context
+
+<!-- The issue motivating this decision, and any context that influences or constrains the decision. -->
+
+Existing Primer ViewComponents behaviours don't use [Catalyst](https://github.github.io/catalyst) or any other dependency, while GitHub developers often start writing a component using Catalyst.
+
+Removing the Catalyst dependency from the Web Component is an extra burden on Web Developers with few benefits. The library is ~1kb of JavaScript, and github.com already includes the library as part of its client-side bundle. Adding the library to Primer ViewComponents does not affect github.com and has a minimal effect on other applications using Primer ViewComponents.
+
+## Decision
+
+<!-- The change that we're proposing or have agreed to implement. -->
+
+Start using Catalyst in Primer ViewComponent behaviours where applicable.
```

Comment:
Would you recommend creating new custom elements for Primer use using catalyst? Is there a preference?

Reply:
I think we should yeah. Should I clarify that somehow in this doc?

Reply:
if there's a preference for catalyst, that would be helpful to make explicit! @lindseywild and I are working on an `ActionMenu` component and a `action-menu` element. we're in the early stages and we're not using catalyst. maybe some point later in the process we should consider changing it to catalyst.. or just start using catalyst for any new element that's introduced